packages/flutter/lib/src/cupertino/activity_indicator.dart

```diff
@@ -128,13 +143,16 @@ class _CupertinoActivityIndicatorPainter extends CustomPainter {
     canvas.save();
     canvas.translate(size.width / 2.0, size.height / 2.0);
 
+    // Rotate so that the top tick is the first to be drawn.
+    canvas.rotate(math.pi / 2);
+
     final int activeTick = (_kTickCount * position.value).floor();
 
-    for (int i = 0; i < _kTickCount; ++ i) {
+    for (int i = ((_kTickCount - 1) * progress).toInt(); i >= 0; --i) {
       final int t = (i + activeTick) % _kTickCount;
-      paint.color = activeColor.withAlpha(_alphaValues[t]);
+      paint.color = activeColor.withAlpha(progress < 1 ? _alphaValues[0] : _alphaValues[t]);
```

Comment:
Maybe mention this color change in the API documentation.

Reply:
I've extracted that out into a private constant to be clearer